src/ngraph/runtime/tensor.cpp

```diff
@@ -70,3 +70,10 @@ void runtime::Tensor::set_stale(bool val)
 {
     m_stale = val;
 }
+
+void runtime::Tensor::copy_to(shared_ptr<runtime::Tensor> d, size_t offset, size_t n)
+{
+    void* copy = malloc(n);
+    read(copy, offset, n);
+    d->write(copy, offset, n);
+}
```

Comment:
free(copy)? The read will be an extra memcpy sometimes. We will have to optimize in the future.

Reply:
Absolutely ! 
I think there might be a few conditions we can optimize for..